doc/introduction.rst

```diff
@@ -0,0 +1,107 @@
+Introduction
+############
+
+`nmigen` is a Python-based tool that aims at automating further the VLSI design process.
+
+`nmigen` makes it possible to apply modern software concepts such as object-oriented programming and metaprogramming to design hardware. This results in more elegant and easily maintained designs and reduces the incidence of human errors.
+
+.. _background:
+
+Background
+**********
+
+Even though the |mm|_ had many successes, it suffers from several limitations stemming from its implementation in manually written Verilog HDL:
+
+.. |mm| replace:: Milkymist System-On-a-Chip
+.. _mm: http://m-labs.hk/m1.html
+
+#. The "event-driven" paradigm of today's dominant hardware descriptions languages (Verilog and VHDL, collectively referred to as "V*HDL" in the rest of this document) is often too general. Today's FPGA architectures are optimized for the implementation of fully synchronous circuits. This means that the bulk of the code for an efficient FPGA design falls into three categories:
+
+   #. Combinatorial statements
+   #. Synchronous statements
+   #. Initialization of registers at reset
+
+   Because V*HDL do not follow this organization, a lot of repetitive manual coding is needed.  This introduces greater opportunity for human error, petty issues, and especially confusion for beginners:
+   
+   #. `wire` vs. `reg` in Verilog
+   #. forgetting to initialize a register at reset
+   #. deciding whether a combinatorial statement must go into a process/always block or not
+   #. simulation mismatches with combinatorial processes/always blocks
+   #. different design strategies for ASICs versus FPGAs
+   #. and more...
+   
+   A little-known fact about FPGAs is that many of them have the ability to initialize their registers from the bitstream contents. This can be done in a portable and standard way using an "initial" block in Verilog, and by affecting a value at the signal declaration in VHDL. This renders an explicit reset signal unnecessary in practice in some cases, which opens the way for further design optimization. However, this form of initialization is entirely not synthesizable for ASIC targets, and it is not easy to switch between the two forms of reset using V*HDL.
```

Comment:
As in, nMigen does not offer the option to switch between the two options either. (It may in the future, but doesn't now.) So this sentence should be removed.

Reply:
How should I resolve this copy then?  I interpret this paragraph not as there being specific ASIC modalities or explicit ASIC support, but rather, "This is a problem contributing to why we wrote nmigen."

Alternatively, I can remove this paragraph all together.